docs/spec/efficient_querying.rst

```diff
@@ -0,0 +1,98 @@
+.. _efficient_querying:
+
+==================
+Efficient Querying
+==================
+
+In general, Kartothek should know as early as possible which data to exclude, so
+Kartothek doesn't have to read and transfer it. If you're going to query a
+dataset a lot, you should consider storing it in an optimized layout. See
+:ref:`writing`.
+
+
+.. _querying_process:
+
+Background
+----------
+
+To understand how to query efficiently, you should understand Kartothek's
+querying process:
+
+When querying data, Kartothek tries to apply the given predicates as early as
+possible, so it has to read and parse as few parquet files as possible:
+
+- Kartothek looks if there exists an index for the filtered column. This may
+  either be if we partitioned the column (primary index) or if we generated a
+  secondary index for it. Kartothek then combines the results for multiple
+  columns.
+- In the resulting set of Parquet files, Kartothek looks at the footer, which
+  states the range of values for each column in this file. If there are multiple
+  row groups, each one specifies the range of values inside it. Kartothek checks
+  if the queried data is in the range of a row group and therefore if this row
+  group is loaded. This process is called Predicate Pushdown.
+
+Kartothek then downloads these row groups and filters the resulting Pandas
```

Comment:
use fetch instead of download. Download somehow feels like we're storing it on the disc but we keep it in memory of course